### PR TITLE
[Core] Move test utility to test file

### PR DESCRIPTION
**Pull request by wjabbour**
## Purpose

This code was introduced about six months ago to shuffle weights into the layout expected by the Triton SwiGLU kernel. ROCm’s AITER operations have [since implemented weight shuffling](https://github.com/wjabbour/vllm/blob/jabbour-remove-dead-function/vllm/_aiter_ops.py#L1816) (e.g. rocm_aiter_ops.shuffle_weights, shuffle_weight_a16w4), and all production code paths use AITER for weight shuffling (mxfp4, quark_moe, fused_moe oracle).

This function in utils.py is not used by any production code. It has been removed from utils.py, and a local shuffle_weight helper has been added to a shared util file for unit tests.

## Test Result

:heavy_check_mark: 

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: tests/kernels/moe/test_gpt_oss_triton_kernels.py [hunk 1]
```diff
@@ -26,10 +26,19 @@
...
 
 
+def shuffle_weight(w: torch.Tensor) -> torch.Tensor:
+    """Fold weights to adjacent locations for Triton MoE kernel layout."""
+    shape = w.shape
+    n = shape[-1]
+    first = w[..., : n // 2]
+    second = w[..., n // 2 :]
+    stacked = torch.stack((first, second), dim=-1)
+    return stacked.reshape(shape)
```
[the remark this answers is not in the document]
Comment (wjabbour) [reply tests/kernels/moe/test_gpt_oss_triton_kernels.py]: done

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!
